app/controllers/api/v1/social_users_controller.rb

```diff
@@ -0,0 +1,55 @@
+module Api
+  module V1
+    class SocialUsersController < ApplicationController
+      before_action :authenticate_user!, only: %i[update]
+
+      # For social media logins, renders the appropriate `redirect_to` path depending on whether the user is registered.
+      #
+      # Requires that the ActionController::Parameters contain a :user key, with a nested :email key.
+      # For example: { user: { email: "john@example.com" } }
+      #
+      # @return [String] A string of the user's redirect_to path
+      # @see https://github.com/zquestz/omniauth-google-oauth2#devise
+      #
+      def show
+        user = User.find_by(email: params.dig(:id))
```

Comment:
As the `email` key is nested in the `user` key, this would be `(email: params.dig(:user, :email))`